packages/build-utils/src/fs/run-user-scripts.ts

```diff
@@ -350,10 +350,17 @@ export async function scanParentDirs(
     start: destPath,
     filename: 'package.json',
   });
-  const packageJson: PackageJson | undefined =
-    readPackageJson && pkgJsonPath
-      ? JSON.parse(await fs.readFile(pkgJsonPath, 'utf8'))
-      : undefined;
+
+  let packageJson: PackageJson | undefined;
+  if (readPackageJson && pkgJsonPath) {
+    try {
+      packageJson = JSON.parse(await fs.readFile(pkgJsonPath, 'utf8'));
+    } catch (err) {
+      throw new Error(
+        `Could not read package.json: ${(err as Error).message}.`
```

Comment:
let's also put the path in the error message?

there may be multiple `package.json` files in a project (which may be read at different times during a build)